lib/gnat/consumer_supervisor.ex

```diff
@@ -189,35 +174,65 @@ defmodule Gnat.ConsumerSupervisor do
     end
   end
 
-  defp initialize_as_manual_consumer(state, connection_pid) do
-    subscriptions = Enum.map(state.subscription_topics, fn(topic_and_queue_group) ->
+  defp subscribe_to_topics(%{service: service}, connection_pid) do
```

Comment:
I generally prefer to do pattern matching in my function heads rather than `if` checks inside the function bodies. I don't feel strongly about this preference, but I like how it looks in this case. What do you think?